modules/dnn/src/onnx/onnx_importer.cpp

```diff
@@ -754,49 +754,35 @@ void ONNXImporter::parseOperatorSet()
     }
 }
 
-void ONNXImporter::handleQuantizedNode(LayerParams& layerParams,
+void ONNXImporter::setParamsDtype(LayerParams& layerParams,
                                        const opencv_onnx::NodeProto& node_proto)
 {
-    // Quantized nodes have output names ending with 'quantized'
-    std::string outName = node_proto.output(0);
-    int len = outName.length();
-    if (len <= 9)
-        return;
-
-    if (outName.substr(len - 9) == "quantized")
+    bool hasVariableInps = false;
+    int i = 0;
+    for (; i < node_proto.input_size(); ++i)
     {
-        outName = outName.substr(0, len - 9);
-        Mat scale, zeropoint;
-
-        if (constBlobs.find(outName + "scale") != constBlobs.end() &&
-            constBlobs.find(outName + "zero_point") != constBlobs.end())
-        {
-            scale = getBlob(outName + "scale");
-            zeropoint = getBlob(outName + "zero_point");
-        }
-        else
+        if (layer_id.find(node_proto.input(i)) != layer_id.end())
         {
-            std::string inpName = node_proto.input(0);
-            inpName = inpName.substr(0, inpName.length() - 9);
-            scale = getBlob(inpName + "scale");
-            zeropoint = getBlob(inpName + "zero_point");
-
-            for (int i = 0; i < node_proto.output_size(); i++)
-            {
-                std::string out = node_proto.output(i);
-                out = out.substr(0, out.length() - 9);
-                addConstant(out + "scale", scale);
-                addConstant(out + "zero_point", zeropoint);
-            }
+            hasVariableInps = true;
+            break;
         }
+    }
 
-        if (scale.total() != 1 || zeropoint.total() != 1)
-            CV_Error(Error::StsNotImplemented, "Per-channel scales/zeropoints are not supported");
+    if (hasVariableInps)
+    {
+        LayerInfo layerInfo = layer_id.find(node_proto.input(i))->second;
```

Comment:
So, you look for any real input layer and check if we created internal layer `*Int8`? I mean, someone could rename the layer or add a new one that doesn't follow that convention. Maybe we're better off with `layerParams.set("depth", CV_8S)` in each layer, or we could create a set of all integer layers. I think this should be discussed further with the rest of the team.

Reply:
Thanks for your code review. OpenCV DNN needs to allocate output blob according to depth type (for example the depth of `CV_8S` will be allocated as int8 Mat.)

> someone could rename the layer or add a new one that doesn't follow that convention.

It is possible. I agree with `create a set of all integer layers`.

Reply:
I'm not sure relying on internal names is a good idea. Let's discuss this on Friday, maybe someone will come up with a better solution.

Reply:
Hi @rogday, please take a look. I have refactored the code.